src/frontend/src/components/dappList.ts

```diff
@@ -0,0 +1,119 @@
+import { html, TemplateResult } from "lit-html";
+
+export type DappDescription = {
+  id: string;
+  name: string;
+  oneLiner?: string;
+  link: string;
+  derivationOrigin?: string;
+  description: string;
+  logo: string;
+};
+
+const createMarqueeListRowItems = (
+  dapps: DappDescription[]
+): TemplateResult => {
+  return html` ${dapps.map((dapp) => {
+    const { logo, name } = dapp;
+
+    return html`<div class="c-marquee__item">
+      <img
+        src="icons/${logo}"
+        alt="${name}"
+        class="c-marquee__image"
+        loading="lazy"
+      />
+    </div>`;
+  })}`;
+};
+
+/**
+ * The marquee is a list of dapps that scrolls horizontally.
+ * It uses a CSS keyframe animation to scroll the each row alternating from left to right and viceversa.
+ * To give the illusion of an infinite scrolling, the list is duplicated horizontally.
+ * when the animation reaches the end of the list, it is reset to the beginning. (default keyframe animation behaviour)
+ */
+
+const createMarqueeList = (dapps: DappDescription[]): TemplateResult => {
+  const itemsPerRow = 5;
+  const totalRows = 4;
+
+  const rows = [];
+
+  for (let i = 0; i < totalRows; i++) {
+    const row = [];
+    for (let j = 0; j < itemsPerRow; j++) {
+      const dapp = dapps[(i * itemsPerRow + j) % dapps.length];
+      row.push(dapp);
+    }
+    rows.push([...row]);
+  }
+
+  // rows are duplicated to create the infinite scrolling effect
+  return html`<div
+    aria-hidden="true"
+    class="c-marquee__list"
+    style="--itemsPerRow: ${itemsPerRow}; --totalRows: ${totalRows}"
+  >
+    ${rows.map(
+      (row, i) => html`<div class="c-marquee__row" style="--rowEq: ${i}">
```

Comment:
what does `rowEq` mean?

Reply:
@nomeata its usually used for CSS selectors. (I started to use it when jQuery did: `https://api.jquery.com/eq-selector/`) But probably index would be better. Il change it.